macro: fix return code (#3022)
metomi/rose: changes.d/3022.fix.md

> The `rose macro` command will now exit 1 in an error scenario it previously exited 0 in.

Reviewer: ```suggestion
The `rose macro` command will now exit 1 instead of 0 in an error scenario.
```
